### PR TITLE
Dataset ini

### PR DESCRIPTION
**Pull request by maxtretiakov**

### CODE REVIEW

File: tools/DatasetExtraction/dataset.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @protected
+    def validate(self):
+        for song in self.songlist:
+            for ch in song:
+                if not isinstance(ch, chord):
+                    raise Exception("Incorrect chord type")
+                    #will also chech cq_max
+
+
+    #returns songlist with Chord class instances
```
Comment (vmkononenko) [inline tools/DatasetExtraction/dataset.py]: I don't think so. IMO `load()` should return a list of lists or a map of lists, with the first layer representing a song, and a second list of segments representing sequence of chord with their start and end timestamps.

File: tools/DatasetExtraction/dataset.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def get_hmm_matrix_i(self):
+        pass
+
+    @private
+    def get_hmm_matrix_t(self):
+        pass
+
+    @private
+    def get_hmm_matrix_e(self):
+        pass
```
Comment (vmkononenko) [inline tools/DatasetExtraction/dataset.py]: there should be public methods actually

File: tools/DatasetExtraction/dataset.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    #i - song, j - chord
+    songlist = [[]]
+
+    @protected
+    def validate(self):
+        for song in self.songlist:
+            for ch in song:
+                if not isinstance(ch, chord):
+                    raise Exception("Incorrect chord type")
+                    #will also chech cq_max
```
Comment (vmkononenko) [inline tools/DatasetExtraction/dataset.py]: this is not what we need from this validation, really. And it should be abstract. For now forget about it and just create an abstract stub.

File: tools/DatasetExtraction/dataset.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+
+    #returns songlist with Chord class instances
+    @protected
+    @abstractmethod
+    def load(self):
+        pass
+
+    @private
+    def get_hmm_matrix_i(self):
```
Comment (vmkononenko) [inline tools/DatasetExtraction/dataset.py:20]: this is the best method to start implementation from. Let's implement it's logic stubbing everything else. Once we have this method you can start working on the rest of stuff.

**Review by vmkononenko — CHANGES_REQUESTED** [review]

**Comment by maxtretiakov** [comment]
Parsing, Ini matrix

**Pull request closed without merging**